website/docs/configuration/types.html.md

```diff
@@ -0,0 +1,187 @@
+  logic.
+
+### Conversion of Primitive Types
+
+The Terraform language will automatically convert `number` and `bool` values
+to `string` values when needed, and vice-versa as long as the string contains
+a valid representation of a number or boolean value.
+
+* `true` converts to `"true"`, and vice-versa
+* `false` converts to `"false"`, and vice-versa
+* `15` converts to `"15"`, and vice-versa
+
+## The "Any" Type
+
+The type keyword `any` is a special type constraint that accepts any value.
+
+## Complex Types
+
+A _complex_ type is a type that groups multiple values into a single value.
+Complex types are represented by type constructor functions, but several of them
+also have shorthand keyword versions.
+
+There are two categories of complex types: collection types (for grouping
+similar values), and structural types (for grouping potentially dissimilar
+values).
+
+### Collection Types
+
+A _collection_ type allows multiple values of _one_ other type to be grouped
+together as a single value. The type of value _within_ a collection is called
+its _element type._ All collection types must have an element type, which is
+provided as the argument to their constructor function.
+
+For example, the type `list(string)` means "list of strings", which is a
+different type than `list(number)`, a list of numbers. All elements of a
+collection must always be of the same type.
+
+The three kinds of collection type in the Terraform language are:
+
+* `list(...)`: a sequence of values identified by consecutive whole numbers
+  starting with zero.
+
+    The type keyword `list` is a shorthand for `list(any)`, which accepts any
+    element type as long as every element is the same type.
+* `map(...)`: a collection of values where each is identified by a string label.
+
+    The type keyword `map` is a shorthand for `map(any)`, which accepts any
+    element type as long as every element is the same type.
+* `set(...)`: a collection of unique values that do not have any secondary
+  identifiers or ordering.
+
+    The type keyword `set` is a shorthand for `set(any)`, which accepts any
+    element type as long as every element is the same type.
+
+### Structural Types
+
+A _structural_ type allows multiple values of _several distinct types_ to be
+grouped together as a single value. Structural types require a _schema_ as an
+argument, to specify which types are allowed for which elements.
+
+The two kinds of structural type in the Terraform language are:
+
+* `object(...)`: a collection of named attributes that each have their own type.
+
+    The schema for object types is `{<KEY> = <TYPE>, <KEY> = <TYPE>, ...}` — a
+    pair of curly braces containing a comma-separated series of `<KEY> = <TYPE>`
+    pairs. Values that match the object type must contain _exactly_ the
+    specified keys (no more and no fewer), and the value for each key must
+    match the specified type for that key.
```

Comment:
I had to infer this, because it was too late in the day to bug you!!! Did I get this correct, or does it behave differently?